In [8]:
# GetOldTweet3 사용 준비
try:
    import GetOldTweets3 as got
except:
    !pip install GetOldTweets3
    import GetOldTweets3 as got


In [9]:
# BeautifulSoup4 사용 준비
try:
    from bs4 import BeautifulSoup as bs
except:
    !pip install bs4
    from bs4 import BeautifulSoup as bs

In [10]:
import datetime 

days_range = [] # datetime 패키지, datetime 클래스(날짜와 시간 함께 저장) # strptime 메서드: 문자열 반환 
start = datetime.datetime.strptime("2020-09-20", "%Y-%m-%d") 
end = datetime.datetime.strptime("2020-09-22", "%Y-%m-%d") 

date_generated = [start+datetime.timedelta(days=x) for x in range(0, (end-start).days)] 

for date in date_generated: 
    days_range.append(date.strftime("%Y-%m-%d")) 

print("===설정된 트윗 수집 기간: {} ~ {}===".format(days_range[0], days_range[-1]))
print("===총 {}일 간의 데이터 수집===".format(len(days_range)))


===설정된 트윗 수집 기간: 2020-09-20 ~ 2020-09-21===
===총 2일 간의 데이터 수집===


In [14]:
import time

start_date = days_range[0]
end_date = (datetime.datetime.strptime(days_range[-1], "%Y-%m-%d") 
            + datetime.timedelta(days=1)).strftime("%Y-%m-%d") # setUntil이 끝을 포함하지 않으므로, day + 1

# tweetCriteria로 수집 기준 정의 
tweetCriteria = got.manager.TweetCriteria().setQuerySearch("경제").setSince("2020-09-20").setUntil("2020-09-22")

In [15]:

print("Collecting data start.. from {} to {}".format(days_range[0], days_range[-1]))
start_time = time.time()

tweet = got.manager.TweetManager.getTweets(tweetCriteria)

print("Collecting data end.. {0:0.2f} Minutes".format((time.time() - start_time)/60))
print("=== Total num of tweets is {} ===".format(len(tweet)))



print("Collecting data end.. {0:0.2f} Minutes".format((time.time() - start_time)/60))
print("=== Total num of tweets is {} ===".format(len(tweet)))

ERROR:root:Internal Python error in the inspect module.
Below is the traceback from this internal error.

An error occured during an HTTP request: HTTP Error 404: Not Found
Try to open in browser: https://twitter.com/search?q=%EA%B2%BD%EC%A0%9C%20since%3A2020-09-20%20until%3A2020-09-22&src=typd
Traceback (most recent call last):
  File "/home/hslee/src/getoldtweets3/GetOldTweets3/manager/TweetManager.py", line 355, in getJsonResponse
    response = opener.open(url)
  File "/home/hslee/anaconda3/lib/python3.8/urllib/request.py", line 531, in open
    response = meth(req, response)
  File "/home/hslee/anaconda3/lib/python3.8/urllib/request.py", line 640, in http_response
    response = self.parent.error(
  File "/home/hslee/anaconda3/lib/python3.8/urllib/request.py", line 569, in error
    return self._call_chain(*args)
  File "/home/hslee/anaconda3/lib/python3.8/urllib/request.py", line 502, in _call_chain
    result = func(*args)
  File "/home/hslee/anaconda3/lib/python3.8/urllib/reque

TypeError: object of type 'NoneType' has no len()

An error occured during an HTTP request: [Errno 104] Connection reset by peer

404에러. 아마도.. 트위터 사이트의 코드자체가 수정된 것으로 보임..